### DISCUSSION TITLE
delete plan and data files on repo

### DISCUSSION THREAD

**Discussion by nocivo72** [garethgeorge/backrest#606] [Q&A] [answered]
I would like to ask if there is a way to remove all files from the repository after deleting a plan. I found the "Forget (Destructive)" button in each snapshot, but if you have many snapshots, it becomes very tedious to apply this action to all of them. Also, I’m not sure if it fully deletes all data from the repository.

Is there a command I can use to delete the data files of a deleted plan? I have some knowledge of Restic, but I’m not very familiar with it. I prefer managing everything directly from Backrest, which I find very user-friendly. is very good

Thank you!

**Comment by garethgeorge** [comment, marked as answer]
Hey, this sort of thing is what the `Run Command` feature aims to support. Easiest way to do this is to get familiar with the restic CLI. https://restic.readthedocs.io/en/stable/060_forget.html#removing-all-snapshots .

In your case, you'll want something like 
 * `forget --tag plan:<your plan ID> --keep-last-n 1 --dry-run ` which previews what would be forgotten but removes nothing -- check the output to be sure everything you expect is getting picked up.
 * `forget --tag plan:<your plan ID> --keep-last-n 1` to actually do the delete.

And then you'll need to manually delete the last snapshot with the tag (restic blocks operations that would delete every snapshot as a safety measure).

It's not a capability I have plans to add to the UI since I think making it a bit difficult reduces the risks of someone in-advertantly clicking a button they don't understand and "nuking" their backups, but I'm open to changing that philosophy.

**Reply by nocivo72** [reply]
So, after running the command you mentioned above, to delete the last snapshot, I need to run the same command but with `--keep-last-n 0`, like this?

`forget --tag plan:<your plan ID> --keep-last-n 0`

Let me know if I'm wrong. I found this in the documentation to delete everything in one go (restic version >= 0.17.0):

`forget --tag plan:<your plan ID> --unsafe-allow-remove-all

Is this correct?

I need this because, during a project, I usually add a new backup that creates a snapshot every hour for the folder I'm working on. Once the project is completed, I delete everything because I already have a daily backup of the entire disk.
